caddyconfig/caddyfile/formatter.go

```diff
@@ -165,6 +168,9 @@ func Format(input []byte) []byte {
 			}
 			continue
 
+		case nesting == 1 && wasComment && ch == '}':
```

Comment:
Are you sure it shouldn't be `nesting > 0`? Could you add a test case with some subdirectives (or a `route`/`handle`) so it's more than one level deep?